packages/application/src/shell.ts

```diff
@@ -609,6 +609,17 @@ export class LabShell extends Widget implements JupyterFrontEnd.IShell {
     this.toggleClass('jp-mod-presentationMode', value);
   }
 
+  /**
+   * Whether JupyterLab is in fullscreen mode with the
+   * `jp-mod-fullscreenMode` CSS class.
+   */
+  get fullscreenMode(): boolean {
+    return this.hasClass('jp-mod-fullscreenMode');
```

Comment:
I don't think we should store state which can be modified by the JS API by using a CSS class - this can lead to inconsistent state if an extension or JS in outputs of user executed code manipulate the fullscreen state (for example rise extension for presenting slides could do that).

I would prefer if the getter returned the state derived from the browser API like `fullscreenElement`. That said, I am not even sure if we need a getter/setter here - if we are not setting the CSS class, then we could get away without extending the public API of `LabShell` at all, which will make merging this PR much easier.

Reply:
We made the changes and removed the getter/setter and directly queried the fullscreenElement value, we also reverted toggle sidebar widget to its original code. Thank you! (: